{
  "threat_severity": "Moderate",
  "public_date": "2020-06-18T00:00:00Z",
  "bugzilla": {
    "description": "CVE-2020-14422 python: DoS via inefficiency in IPv{4,6}Interface classes",
    "id": "1854926",
    "url": "https://bugzilla.redhat.com/show_bug.cgi?id=1854926"
  },
  "cvss3": {
    "cvss3_base_score": "5.9",
    "cvss3_scoring_vector": "CVSS:3.1/AV:N/AC:H/PR:N/UI:N/S:U/C:N/I:N/A:H",
    "status": "draft"
  },
  "cwe": "CWE-400",
  "details": [
    "Lib/ipaddress.py in Python through 3.8.3 improperly computes hash values in the IPv4Interface and IPv6Interface classes, which might allow a remote attacker to cause a denial of service if an application is affected by the performance of a dictionary containing IPv4Interface or IPv6Interface objects, and this attacker can cause many dictionary entries to be created.",
    "A vulnerability was found in the way the ipaddress python module computes hash values in the IPv4Interface and IPv6Interface classes. This flaw allows an attacker to create many dictionary entries, due to the performance of a dictionary containing the IPv4Interface or IPv6Interface objects, possibly resulting in a denial of service. The highest threat from this vulnerability is to system availability."
  ],
  "statement": "In Red Hat Enterprise Linux, python3 includes the ipaddress module by default, while for python2 a separate package, python-ipaddress, needs to be installed for the module to be used. Moreover, the ipaddress module is included in other packages as well, like python-pip.",
  "package_state": [
    {
      "product_name": "Red Hat Enterprise Linux 5",
      "fix_state": "Out of support scope",
      "package_name": "python",
      "cpe": "cpe:/o:redhat:enterprise_linux:5"
    },
    {
      "product_name": "Red Hat Enterprise Linux 6",
      "fix_state": "Out of support scope",
      "package_name": "python",
      "cpe": "cpe:/o:redhat:enterprise_linux:6"
    },
    {
      "product_name": "Red Hat Enterprise Linux 7",
      "fix_state": "Affected",
      "package_name": "python3",
      "cpe": "cpe:/o:redhat:enterprise_linux:7"
    },
    {
      "product_name": "Red Hat Enterprise Linux 7",
      "fix_state": "Affected",
      "package_name": "python-ipaddress",
      "cpe": "cpe:/o:redhat:enterprise_linux:7"
    },
    {
      "product_name": "Red Hat Enterprise Linux 7",
      "fix_state": "Affected",
      "package_name": "python-pip",
      "cpe": "cpe:/o:redhat:enterprise_linux:7"
    },
    {
      "product_name": "Red Hat Enterprise Linux 8",
      "fix_state": "Affected",
      "package_name": "python27:2.7/python2-pip",
      "cpe": "cpe:/o:redhat:enterprise_linux:8"
    },
    {
      "product_name": "Red Hat Enterprise Linux 8",
      "fix_state": "Affected",
      "package_name": "python27:2.7/python-ipaddress",
      "cpe": "cpe:/o:redhat:enterprise_linux:8"
    },
    {
      "product_name": "Red Hat Enterprise Linux 8",
      "fix_state": "Affected",
      "package_name": "python3",
      "cpe": "cpe:/o:redhat:enterprise_linux:8"
    },
    {
      "product_name": "Red Hat Enterprise Linux 8",
      "fix_state": "Not affected",
      "package_name": "python36:3.6/python36",
      "cpe": "cpe:/o:redhat:enterprise_linux:8"
    },
    {
      "product_name": "Red Hat Enterprise Linux 8",
      "fix_state": "Affected",
      "package_name": "python38:3.8/python38",
      "cpe": "cpe:/o:redhat:enterprise_linux:8"
    },
    {
      "product_name": "Red Hat Enterprise Linux 8",
      "fix_state": "Affected",
      "package_name": "python38:3.8/python3x-pip",
      "cpe": "cpe:/o:redhat:enterprise_linux:8"
    },
    {
      "product_name": "Red Hat Enterprise Linux 8",
      "fix_state": "Affected",
      "package_name": "python-pip",
      "cpe": "cpe:/o:redhat:enterprise_linux:8"
    },
    {
      "product_name": "Red Hat Quay 3",
      "fix_state": "Affected",
      "package_name": "python-ipaddress",
      "cpe": "cpe:/a:redhat:quay:3::el7"
    },
    {
      "product_name": "Red Hat Software Collections",
      "fix_state": "Affected",
      "package_name": "python27-python-pip",
      "cpe": "cpe:/a:redhat:rhel_software_collections:3"
    },
    {
      "product_name": "Red Hat Software Collections",
      "fix_state": "Affected",
      "package_name": "rh-python36-python",
      "cpe": "cpe:/a:redhat:rhel_software_collections:3"
    },
    {
      "product_name": "Red Hat Software Collections",
      "fix_state": "Affected",
      "package_name": "rh-python36-python-pip",
      "cpe": "cpe:/a:redhat:rhel_software_collections:3"
    },
    {
      "product_name": "Red Hat Software Collections",
      "fix_state": "Affected",
      "package_name": "rh-python38-python",
      "cpe": "cpe:/a:redhat:rhel_software_collections:3"
    },
    {
      "product_name": "Red Hat Software Collections",
      "fix_state": "Affected",
      "package_name": "rh-python38-python-pip",
      "cpe": "cpe:/a:redhat:rhel_software_collections:3"
    }
  ],
  "name": "CVE-2020-14422",
  "mitigation": {
    "value": "As a short term solution, if your application is using the IPv4Interface/IPv6Interface classes as keys of a dictionary, it is possible to patch the __hash__ method of those classes to not be constant.\n```\nIPv4Interface.__hash__ = lambda self: hash((self._ip, self._prefixlen, int(self.network.network_address)))\nIPv6Interface.__hash__ = lambda self: hash((self._ip, self._prefixlen, int(self.network.network_address)))\n```",
    "lang": "en:us"
  }
}
